### PR TITLE
use the etcd observers from library-go

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

File: pkg/operator/configobservation/configobservercontroller/observe_config_controller.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func NewConfigObserver(
 				SecretLister_:                kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Lister(),
 				ConfigSecretLister_:          kubeInformersForNamespaces.InformersFor(operatorclient.GlobalUserSpecifiedConfigNamespace).Core().V1().Secrets().Lister(),
 				OpenshiftEtcdEndpointsLister: kubeInformersForNamespaces.InformersFor("openshift-etcd").Core().V1().Endpoints().Lister(),
-				ConfigmapLister:              kubeInformersForNamespaces.InformersFor("openshift-etcd").Core().V1().ConfigMaps().Lister(),
+				ConfigmapLister_:             kubeInformersForNamespaces.ConfigMapLister(),
```
Comment (p0lyn0mial) [inline pkg/operator/configobservation/configobservercontroller/observe_config_controller.go:90]: required by the recent change in the cloud provider observer - https://github.com/openshift/library-go/commit/16d6830d0b80dc2a3315207116d009ed2dd4cebf#diff-9a42c46842caaad30adbbb0b5557cda9

**Comment by p0lyn0mial** [comment]
/hold 

needs https://github.com/openshift/library-go/pull/839 first

File: pkg/operator/configobservation/etcdendpoints/observe_etcd_endpoints_test.go [hunk 2]
```diff
@@ -117,12 +117,23 @@ func TestObserveStorageURLs(t *testing.T) {
...
+			expected: observedConfig(withStorageURL("https://10.0.0.1:2379"), withLocalhostStorageURLs()),
+		},
+		{
+			name:         "NoCurrentConfigNoEndPoints",
+			endpoint:     endpoints(),
+			expected:     observedConfig(withLocalhostStorageURLs()),
+			expectErrors: true,
+		},
 	}
 	for _, tt := range tests {
```
Comment (p0lyn0mial) [inline pkg/operator/configobservation/etcdendpoints/observe_etcd_endpoints_test.go:132]: I think that I'm going to leave these tests here and maybe I will move a subset to lib-go.

File: pkg/operator/configobservation/etcdendpoints/observe_etcd_endpoints.go [hunk 3]
```diff
@@ -1,105 +1,37 @@
...
 // then the previously observed storage config URLs will be re-used.
 func ObserveStorageURLs(genericListers configobserver.Listers, recorder events.Recorder, currentConfig map[string]interface{}) (map[string]interface{}, []error) {
-	listers := genericListers.(configobservation.Listers)
-	oldStorageConfigURLsPath := []string{"storageConfig", "urls"}
-	newStorageConfigURLsPath := []string{"apiServerArguments", "etcd-servers"}
 	var errs []error
 
 	previouslyObservedConfig := map[string]interface{}{}
-	oldCurrentEtcdURLs, oldFound, err := unstructured.NestedStringSlice(currentConfig, oldStorageConfigURLsPath...)
+	oldCurrentEtcdURLs, oldFound, err := unstructured.NestedStringSlice(currentConfig, libgoetcd.OldStorageConfigURLsPath...)
```
Comment (deads2k) [inline pkg/operator/configobservation/etcdendpoints/observe_etcd_endpoints.go:17]: The library-go code has knowledge of these paths, why do we need to have this stanza here?

**Comment by deads2k** [comment]
/approve

I'd like to see a proof PR to the openshift-apiserver too.

**Pull request closed without merging**